queue/src/fiskaltrust.Middleware.Localization.QueueDE/Repositories/ReadOnlyReceiptReferenceRepository.cs

```diff
@@ -124,19 +125,23 @@ private async Task<bool> AddReference(HashSet<ReceiptReferencesGroupedData> rece
 
         private async Task<(long, DateTime?)> GetLastZNumberForQueueItem(ftQueueItem queueItem)
         {
-            var actionJournals = (await _actionJournalRepository.GetAsync()).Where(x => x.Type == "4445000000000007").OrderBy(x => x.TimeStamp);
+            var regexDaily = new Regex("4445[0-9]{11}7");
```

Comment:
```suggestion
            var regexDaily = new Regex("4445[0-9]{8}0007");
```
I think technically the last 4 numbers are part of the receipt case. And we also have `0x4445000000000017` (init-scu-switch) right now which would be caught wrongly with the current regex